src/psyclone/core/component_indices.py

```diff
@@ -116,8 +116,16 @@ def __getitem__(self, indx):
         :rtype: list of :py:class:`psyclone.psyir.nodes.Node`, or \
             :py:class:`psyclone.psyir.nodes.Node`
 
+        :raises IndexError:
```

Comment:
Please complete :-)

Reply:
Now that is really embarrassing (well, I originally didn't add detailed error messages, so I added this, then tests, then pylint, then ... forgot about finishing it :( )